### PR TITLE
Fix some macOS warnings and arm atomic

### PR DESCRIPTION
**Pull request by adizaimi**
* Fix some macOS warnings
* Arm would not link without -latomic anymore
* Get test tools to compile without errors
* Fix readlink issues in macOS, so now tests can run

### CODE REVIEW

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/prefault_helper.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/sqlglue.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/handle_buf.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c40179a/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>1/374 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>socksql_master_swings</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c40179a/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c40179a/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2532'>Resubmit</a></p>

File: db/sqlinterfaces.c [hunk 1]
```diff
@@ -4814,7 +4814,7 @@ static int prepare_engine(struct sqlthdstate *thd, struct sqlclntstate *clnt,
 
     /* Do this here, before setting up Btree structures!
        so we can get back at our "session" information */
-    clnt->debug_sqlclntstate = pthread_self();
+    clnt->debug_sqlclntstate = (int)pthread_self();
```
Comment (akshatsikarwar) [inline db/sqlinterfaces.c:4817]: Change `debug_sqlclntstate` to `pthread_t` instead?

File: lua/sp.c [hunk 2]
```diff
@@ -2131,11 +2131,11 @@ static void InstructionCountHook(lua_State *lua, lua_Debug *debug)
                         sp->spversion.version_str, debug->currentline);
             }
             if (strcasecmp(sp_info, gbl_break_spname) == 0) {
-                gbl_break_lua = pthread_self();
+                gbl_break_lua = (int)pthread_self();
```
Comment (akshatsikarwar) [inline lua/sp.c:2134]: Same - declare `gbl_break_lua` as `pthread_t`

File: lua/sp.c [hunk 3]
```diff
@@ -5848,7 +5850,7 @@ static void debug_sp(struct sqlclntstate *clnt)
 halt_here:
     debug_clnt = clnt;
     if (arg1 == 0) {
-        gbl_break_lua = INT_MAX;
+        gbl_break_all_lua = 1;
```
Comment (mohitkhullar) [inline lua/sp.c:5853]: The lua part looks good.

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/prefault_helper.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/sqlglue.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/handle_buf.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/d9c4e7a/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>1/375 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_swapfields_logicalsc_generated</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/d9c4e7a/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/d9c4e7a/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2532'>Resubmit</a></p>

**Pull request merged**